crates/api/src/local_user.rs

```diff
@@ -164,11 +178,37 @@ impl Perform for SaveUserSettings {
 
     let avatar = diesel_option_overwrite_to_url(&data.avatar)?;
     let banner = diesel_option_overwrite_to_url(&data.banner)?;
-    let email = diesel_option_overwrite(&data.email.clone().map(Sensitive::into_inner));
     let bio = diesel_option_overwrite(&data.bio);
     let display_name = diesel_option_overwrite(&data.display_name);
     let matrix_user_id = diesel_option_overwrite(&data.matrix_user_id);
     let bot_account = data.bot_account;
+    let email_deref = data.email.as_deref().map(|e| e.to_owned());
+
+    let email = if let Some(email) = &email_deref {
+      let site = blocking(context.pool(), Site::read_simple).await??;
+      if site.require_email_verification {
```

Comment:
This is also how i implemented it, but i dont think it makes sense. If there is an email being set, Lemmy should always send a verification mail, regardless of the `require_email_verification` setting. Cause its really useless to have an email with a typo in it. What `require_email_verification` should do is make the email field mandatory on signup (and prevent unsetting it in the user settings), nothing else.